core-bundle/src/Resources/contao/controllers/FrontendIndex.php

```diff
@@ -263,23 +265,21 @@ public function renderPage($pageModel)
 		// Authenticate the user if the page is protected
 		if ($objPage->protected)
 		{
-			$container = System::getContainer();
-			$token = $container->get('security.token_storage')->getToken();
+			$security = System::getContainer()->get('security.helper');
 
-			if ($container->get('security.authentication.trust_resolver')->isAnonymous($token))
+			if (!$security->isGranted(ContaoCorePermissions::MEMBER_IN_GROUPS, $objPage->groups))
 			{
-				throw new InsufficientAuthenticationException('Not authenticated: ' . Environment::get('uri'));
-			}
+				if ($security->getToken() instanceof AnonymousToken)
```

Comment:
This was previously `$container->get('security.authentication.trust_resolver')->isAnonymous($token)`. Guess it does not matter, does it?